classes/submission/SubmissionDAO.inc.php

```diff
@@ -322,21 +321,28 @@ function getAssignedById($submissionId, $userId, $contextId = null, $useCache =
 			unset($submission);
 		}
 
-		$params = $this->_getFetchParameters();
-		$params[] = (int) $submissionId;
-		$params[] = (int) $userId;
+		$params = array_merge(
+			array((int) ROLE_ID_AUTHOR),
+			$this->_getFetchParameters(),
+			array((int) $submissionId)
+		);
 		if ($contextId) $params[] = (int) $contextId;
 
 		$result = $this->retrieve(
-				'SELECT	s.*, ps.date_published,
+			'SELECT	s.*, ps.date_published,
 				' . $this->_getFetchColumns() . '
-				FROM	submissions s
+			FROM	submissions s
 				LEFT JOIN published_submissions ps ON (s.submission_id = ps.submission_id)
+				LEFT JOIN stage_assignments asa ON (asa.submission_id = s.submission_id)
+				LEFT JOIN user_groups aug ON (asa.user_group_id = aug.user_group_id AND aug.role_id = ?)
 				' . $this->_getFetchJoins() . '
-				WHERE	s.submission_id = ?
-				AND s.date_submitted IS NOT NULL AND ps.date_published IS NULL AND s.user_id <> ? AND s.status <> ' .  STATUS_DECLINED .
```

Comment:
with this change if an user is assigned as author, his submission will not appear in the assigned submissions list anymore right? It will now appear in the my submissions grid. I think that previously this was different. My submissions was only the one that had the same user_id (the one that submitted originally). But that makes more sense now.